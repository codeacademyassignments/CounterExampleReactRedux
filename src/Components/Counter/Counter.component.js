import React from 'react';
import PropTypes from 'prop-types';
import Button from '../Button/Button.component';

// class Counter extends Component{
// constructor(props){
//   super(props);
//   console.log('conso');
// }


// static getDerivedStateFromProps(props,state)
// {
//   console.log('done',props);
//   return state;
// }
// state = {
//   count: this.props.initialValue,
// }
// increment = ()=>{
//   this.setState({count: this.state.count+1});
// }
// decrement = ()=>{
//   console.log('minus');
//   this.setState({count:this.state.count-1});
// }
const Counter = ({ counterValue, incrementCounter, decrementCounter }) => (
  <div>
    <p className="counterValue">
Counter:
      {counterValue}
    </p>
    <div>
      <Button buttonText="ADD" onClick={incrementCounter} />
      <Button buttonText="MINUS" onClick={decrementCounter} />
    </div>
  </div>
);

Counter.propTypes = {
  counterValue: PropTypes.number.isRequired,
  incrementCounter: PropTypes.func.isRequired,
  decrementCounter: PropTypes.func.isRequired,
};
export default Counter;
