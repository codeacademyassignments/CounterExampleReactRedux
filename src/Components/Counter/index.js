import React , {Component} from 'react';
import Button from '../Button';

class Counter extends Component{

  state = {
    count: this.props.initialValue,
  }
  increment = ()=>{
    // console.log('somethign');
    // console.log(this.state.count);
    //this.state.count = this.state.count+1;
    this.setState({count: this.state.count+1});
  }
  decrement = ()=>{
    console.log('minus');
    this.setState({count:this.state.count-1});
  }
  render(){
    
    //const counter =10;
    return (
      <div>
        <p>Counter:{this.state.count}</p>
      <div>
      <Button buttonText = "ADD" onClick={this.increment}/>
      <Button buttonText = "MINUS" onClick={this.decrement}/>
</div>
</div>
    )
}
}

export default Counter