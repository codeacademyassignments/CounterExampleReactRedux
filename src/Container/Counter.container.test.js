import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import CounterContainer from './Counter.container';
import Counter from '../Components/Counter/Counter.component';

const mockDispatch = jest.fn();

const storeFake = passedState => ({
  default: jest.fn(),
  subscribe: jest.fn(),
  dispatch: mockDispatch,
  getState: () => passedState,
});
const store = storeFake({});

describe('Counter Container', () => {
  it('Snapshot should match with old snap', () => {
    const render = renderer.create(<Provider store={store}><CounterContainer /></Provider>);
    expect(render).toMatchSnapshot();
  });

  describe('mapDispatchToProps', () => {
    const wrapper = mount(<Provider store={store}><CounterContainer /></Provider>);
    const counterContainer = wrapper.find(CounterContainer);
    const counterComponent = counterContainer.find(Counter);
    xit('should dispatch incrementCounter action when incrementCounter called', () => {
      // const wrapper = shallow(<Provider store={store}><CounterContainer /></Provider>);
      wrapper.mapDispatchToProps(mockDispatch).incrementCounter();
      expect(mockDispatch).toHaveBeenCalledWith('incrementCounter()');
    });
    xit('should dispatch decrementCounter action when decrementCounter called', () => {
      // const wrapper = shallow(<Provider store={store}><CounterContainer /></Provider>);
      wrapper.instance().mapDispatchToProps().incrementCounter();
      expect(mockDispatch).toHaveBeenCalledWith('incrementCounter()');
    });
    it('CounterContainer and Counter both should be rendered', () => {
      expect(counterContainer.length).toBeTruthy();
      expect(counterComponent.length).toBeTruthy();
    });
    it('mapStateToProps: should map state to props', () => {
      const expectedAttributes = ['incrementCounter', 'decrementCounter'];
      expect(Object.keys(counterComponent.props())).toEqual(expect.arrayContaining(expectedAttributes));
    });
    it('mapDispatchToProps: should map dispatch to props', () => {
      const expectedAttributes = ['counterValue'];
      expect(Object.keys(counterComponent.props())).toEqual(expect.arrayContaining(expectedAttributes));
    });
  });
});
