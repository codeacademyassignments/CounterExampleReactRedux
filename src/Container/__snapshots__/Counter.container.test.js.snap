// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Container Snapshot should match with old snap 1`] = `
<div>
  <p
    className="counterValue"
  >
    Counter:
  </p>
  <div>
    <button
      className="style"
      onClick={[Function]}
      style={
        Object {
          "color": "black",
        }
      }
      type="button"
    >
      ADD
    </button>
    <button
      className="style"
      onClick={[Function]}
      style={
        Object {
          "color": "black",
        }
      }
      type="button"
    >
      MINUS
    </button>
  </div>
</div>
`;
